modules/ui/rapid_internal_features_dialog.js

```diff
@@ -0,0 +1,153 @@
+
+        modalSelection.select('.modal')
+            .attr('class', 'modal-splash modal modal-rapid');
+
+        var modal = modalSelection.select('.content')
+            .append('form')
+            .attr('class', 'fillL rapid-feature internal rapid-stack')
+
+        modal   
+            .append('div')
+            .attr('class','modal-heading rapid')
+            .html(t('rapid_internal_features.top_label', {
+                icon: icon('#iD-worldailogo', 'logo-rapid'),
+            })); 
+
+        modal   
+            .append('div')
+            .attr('class','modal-heading-desc rapid')
+            .html(t('rapid_internal_features.top_label_desc'));
+
+        modal
+            .append('div')
+            .attr('class','modal-section rapid-checkbox section-divider strong');
+
+
+        addCheckBox({
+            modal: modal, 
+            id: 'rapid-internal-features-ai-halo',
+            label: t('rapid_internal_features.ai_feature_halo'), 
+            description: t('rapid_internal_features.ai_feature_halo_desc'),
+            handler: toggleAiFeatureHalo,
+            enabled: featureEnabled('aiFeatureHalo'),
+            greyout: false,
+            imgid: 'ai-feature-halo', 
+        });
+    
+        modal
+            .append('div')
+            .attr('class','modal-section rapid-checkbox section-divider');
+
+        addCheckBox({
+            modal: modal, 
+            id: 'rapid-internal-features-tagnostic-combine',
+            label: t('rapid_internal_features.tagnostic_road_combine'), 
+            description: t('rapid_internal_features.tagnostic_road_combine_desc'),
+            handler: toggleTagnosticRoadCombine,
+            enabled: featureEnabled('tagnosticRoadCombine'),
+            greyout: false,
+            imgid: 'tagnostic-road-combine', 
+        }); 
+
+        modal
+            .append('div')
+            .attr('class','modal-section rapid-checkbox section-divider');
+
+        addCheckBox({
+            modal: modal, 
+            id: 'rapid-internal-features-tag-sources',
+            label: t('rapid_internal_features.tag_sources'), 
+            description: t('rapid_internal_features.tag_sources_desc'),
+            handler: toggleTagSources,
+            enabled: featureEnabled('tagSources'),
+            greyout: false,
+            imgid: 'tag-sources', 
+        }); 
+    };
+
+
+    function addCheckBox(options) {
+        var toggleOption = options.modal
+            .append('div')
+            .attr('class','modal-section rapid-checkbox internal')
+            .classed('disabled', options.greyout)
+            .attr('id', 'section-' + options.id)
+
+        var toggleOptionContainer =  toggleOption.append('div')
+            .attr('class', 'rapid-feature-label-container internal'); 
+
+        toggleOptionContainer
+            .append('div')
+            .attr('id', options.imgid)
+            .attr('class', 'rapid internal feature animation');
```

Comment:
That's a lot of generic words as CSS class names! Can we invert this? Like rapid-internal at the top-level or something? And then we can use `feature` after that, if it's a normal iD way of doing these things. 

We can sync on this. Generally spreading generic names around as CSS classes will just bite you later when  you have conflicts. Specific names are easier to target and maintain in CSS's global namespace.

Reply:
The reason for the code discrepancy between this feature vs. the other two is the other two only manifest their behavior after the user does something- i.e. a specific interaction with iD. 

This feature affects the runtime of iD insofar as it changes the way things are styled on the map. this was my way of wiring things up so that the setting would be read initially before the first paint. That said, looking at this code it is jarring to see one setting getting what appears to be 'preferential' treatment from the others.

Reply:
Fixed to use 'rapid-feature-animation' here.